projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.html

```diff
@@ -1,29 +1,22 @@
-      [displayValue]="displayValue"></kui-text-value-as-string>
-    <kui-int-value #displayVal *ngSwitchCase="constants.IntValue" [mode]="mode" [displayValue]="displayValue">
-    </kui-int-value>
-    <kui-boolean-value #displayVal *ngSwitchCase="constants.BooleanValue" [mode]="mode" [displayValue]="displayValue">
-    </kui-boolean-value>
-    <kui-uri-value #displayVal *ngSwitchCase="constants.UriValue" [mode]="mode" [displayValue]="displayValue">
-    </kui-uri-value>
-    <kui-decimal-value #displayVal *ngSwitchCase="constants.DecimalValue" [mode]="mode" [displayValue]="displayValue">
-    </kui-decimal-value>
-    <kui-color-value #displayVal *ngSwitchCase="constants.ColorValue" [mode]="mode" [displayValue]="displayValue">
-    </kui-color-value>
-    <kui-interval-value #displayVal *ngSwitchCase="constants.IntervalValue" [mode]="mode" [displayValue]="displayValue">
-    </kui-interval-value>
-    <kui-time-value #displayVal *ngSwitchCase="constants.TimeValue" [mode]="mode" [displayValue]="displayValue">
-    </kui-time-value>
+  <span [ngSwitch]="valueTypeOrClass">
+    <kui-text-value-as-string #displayVal *ngSwitchCase="'ReadTextValueAsString'" [mode]="mode" [displayValue]="displayValue"></kui-text-value-as-string>
+    <kui-text-value-as-html #displayVal *ngSwitchCase="'ReadTextValueAsHtml'" [mode]="mode" [displayValue]="displayValue"></kui-text-value-as-html>
+    <kui-int-value #displayVal *ngSwitchCase="constants.IntValue" [mode]="mode" [displayValue]="displayValue"></kui-int-value>
+    <kui-boolean-value #displayVal *ngSwitchCase="constants.BooleanValue" [mode]="mode" [displayValue]="displayValue"></kui-boolean-value>
+    <kui-uri-value #displayVal *ngSwitchCase="constants.UriValue" [mode]="mode" [displayValue]="displayValue"></kui-uri-value>
+    <kui-decimal-value #displayVal *ngSwitchCase="constants.DecimalValue" [mode]="mode" [displayValue]="displayValue"></kui-decimal-value>
+    <kui-color-value #displayVal *ngSwitchCase="constants.ColorValue" [mode]="mode" [displayValue]="displayValue"></kui-color-value>
+    <kui-interval-value #displayVal *ngSwitchCase="constants.IntervalValue" [mode]="mode" [displayValue]="displayValue"></kui-interval-value>
+    <kui-time-value #displayVal *ngSwitchCase="constants.TimeValue" [mode]="mode" [displayValue]="displayValue"></kui-time-value>
     <kui-link-value #displayVal *ngSwitchCase="constants.LinkValue" [mode]="mode" [displayValue]="displayValue"
                     [parentResource]="parentResource" [propIri]="displayValue.property"></kui-link-value>
   </span>
 </div>
 <div>
-  <button class="edit" *ngIf="canModify && !editModeActive" (click)="activateEditMode()">edit</button>
-  <button class="save" *ngIf="editModeActive" [disabled]="!displayValueComponent.form.valid"
-    (click)="saveEditValue()">Save</button>
-  <button class="cancel" *ngIf="editModeActive" (click)="cancelEditValue()">Cancel</button>
+  <span *ngIf="valueTypeOrClass !== 'ReadTextValueAsHtml'">
```

Comment:
maybe change this to check for a "readonly" boolean (ex: readonlyProp) in the event that we have more readonly properties in the future which can be set in the `getValueTypeOrClass` method

Reply:
So you suggest to evaluate this in a method and then just check for a boolean in the template? I think that's a good idea, will do.

Reply:
exactly!

Reply:
done in https://github.com/dasch-swiss/knora-ui-ng-lib/pull/55/commits/a3d215f8f7442b1414b3af0d2c563b97f338a68b